core/shared/src/main/scala/org/typelevel/log4cats/PagingSelfAwareStructuredLogger.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018 Christopher Davenport
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+
+import cats._
+import cats.implicits._
+
+import java.util.UUID
+
+object PagingSelfAwareStructuredLogger {
+
+  /**
+   * Wrap a SelfAwareStructuredLogger adding pagination functionality.
+   *
+   * @param pageSizeK The size (unit is kilobyte) of the chunk of message in each page; this does not include the
+   *                  page header and footer, and tracing data.
+   * @param logger The SelfAwareStructuredLogger to be used to do the actual logging.
+   * @tparam F Effect type class.
+   * @return SelfAwareStructuredLogger with pagination.
+   */
+  def withPaging[F[_]: Monad](pageSizeK: Int, maxPageNeeded: Int)(
```

Comment:
Sorry Ross, I was busy on other issues and could not respond to your comments earlier. The motivation of using K as unit of page size is to make it easier to determine the value, the reason we want to do paging is because the underline logging system has some size limit on the a single log entry, some will split a large log entry into multiple entries, but cut off the logging context, by setting the page size below that size limit you can avoid splitting by logging system; usually the log size limit should be in the order of KB.
Adding scaladoc for  maxPageNeeded.

Reply:
Will add unit tests.